src/test/scala/com/eclipsesource/schema/SchemaValidatorSpec.scala

```diff
@@ -165,39 +167,37 @@ class SchemaValidatorSpec extends PlaySpecification {
       }
     }
 
-  "should resolve references on the classpath via UrlResolver" in {
-    val validator = SchemaValidator().addUrlResolver(ClasspathUrlResolver())
+  "should resolve references on the classpath with ClasspathUrlResolver" in {
+    val validator = SchemaValidator().addUrlResolver(ClasspathUrlProtocolHandler())
     // some.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve references on the classpath via UrlResolver and custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
-    // some.json references location.json within JAR
+      .addRelativeUrlHandler(new ClasspathHandler)
+    // some-issue-65.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some-issue-65.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve relative references with custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65 - 2)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
+      .addRelativeUrlHandler(new ClasspathHandler)
     val mySchemaJson = getClass.getResourceAsStream("/my-schema.schema")
     val mySchema = JsonSource.schemaFromStream(mySchemaJson)
     val result = validator.validate(mySchema.get, Json.obj("foo" -> Json.obj("bar" -> "whatever")))
     result.isSuccess must beTrue
   }
 
-  "should fail resolving relative references with custom protocols if option not set(#65)" in {
+  "should fail resolving relative references without any relative URL handler available (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
+      .addUrlResolver(ClasspathUrlProtocolHandler())
```

Comment:
Yeah, I think some examples are needed, since I don't think I did a good job of being clear. I was really thinking that you would have only one additional classpath handler (a protocol-less version), so that you've got only 2 total:

1. `ClasspathUrlHandler` which extends `URLStreamHandler` (i.e. no protocol)
2. `ClasspathUrlProtocolHandler` which extends `UrlProtocolHandler` which extends `URLStreamHandler` (protocol = "classpath")

Say you've got the following files in your jar's resources:

`/schemas/my_schema.schema`:
```
{
  "properties": {
    "foo": {
      "$ref": <foo schema path>
    }
  }
}
```

`/schemas/foo.schema`:
```
{
  "properties": {
    "bar": {
      "type": "string"
    }
  }
}
```

Now, the user has a few options (not including the case of using **both** relative and absolute simultaneously here):

## Absolute paths
### Protocol-less
`<foo schema path>` = `"/schemas/foo.schema#"`
Use `SchemaValidator().addUrlHandler(ClasspathUrlHandler())`

### Protocol-ful
`<foo schema path>` = `"classpath://schemas/foo.schema#"`
Use `SchemaValidator().addUrlHandler(ClasspathUrlProtocolHandler())`

## Relative paths (relative to directory of top-level schema)
### Protocol-less
`<foo schema path>` = `"foo.schema#"`
Use `SchemaValidator().addRelativeUrlHandler(ClasspathUrlHandler())`

### Protocol-ful
`<foo schema path>` = `"classpath://foo.schema#"`
Use `SchemaValidator().addRelativeUrlHandler(ClasspathUrlProtocolHandler())`

**Note:** I know that the first option is not possible at the moment, due to the work it would take, but this is just a vision (you would have `addUrlHandler(handler: UrlProtocolHandler)` instead of the `addUrlHandler(handler: URLStreamHandler)` that you have right now).

See how the protocol and the relative-ness are decoupled, in that you can choose to have a protocol or not for either absolute or relative paths? You would just need to do some thinking about the order of trying the various set of handlers (I haven't thought it through yet).